documentation/jetty/modules/code/examples/src/main/java/org/eclipse/jetty/docs/programming/JSONDocs.java

```diff
@@ -0,0 +1,206 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.docs.programming;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jetty.util.ajax.AsyncJSON;
+import org.eclipse.jetty.util.ajax.JSON;
+
+@SuppressWarnings("unused")
+public class JSONDocs
+{
+    public void sync()
+    {
+        // tag::sync[]
+        // Full JSON document.
+        String json = """
+            {
+              "field": "value",
+              "count": 42,
+              "array": ["one", "two"],
+              "object": {
+                "size": 13
+              }
+            }
+            """;
+        JSON parser = new JSON();
+        Object object = parser.parse(new JSON.StringSource(json));
+        // The object can be cast to a Map, given the JSON document structure.
+        Map<String, Object> map = (Map<String, Object>)object;
+        // end::sync[]
+    }
+
+    public void async()
+    {
+        // tag::async[]
+        AsyncJSON parser = new AsyncJSON.Factory().newAsyncJSON();
+
+        String jsonChunk1 = """
+            {
+              "field": "value",
+              "cou""";
+        parser.parse(StandardCharsets.UTF_8.encode(jsonChunk1));
+
+        String jsonChunk2 = """
+            nt": 42,
+              "array": ["one", "two"],
+              "object": {
+                "size": 13
+              }
+            }
+            """;
+        parser.parse(StandardCharsets.UTF_8.encode(jsonChunk2));
```

Comment:
I think it would be clearer as:

```suggestion
        String json = """
            {
              "field": "value",
              "count": 42,
              "array": ["one", "two"],
              "object": {
                "size": 13
              }
            }
            """;
            
        parser.parse(StandardCharsets.UTF_8.encode(json.substring(0,20));
        parser.parse(StandardCharsets.UTF_8.encode(json.substring(20));
```

Reply:
Instead of `String jsonChunk`, I would store the result of `StandardCharsets.UTF_8.encode()` in a variable  as I suspect few will remember the return type of that method from the top of their head, and it would make it clear what type is expected by `Parser.parse()`.

Reply:
@gregw No, I really would like to avoid having the whole JSON in a single string.
I added clarifying comments about the two chunks simulating a network read.